src/app/public/modules/sky-tree-view/sky-tree-view-toolbar.component.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  TreeModel,
+  TreeNode
+} from 'angular-tree-component';
+
+import {
+  SkyTreeViewOptions
+} from './types/sky-tree-view-options';
+
+@Component({
+  selector: 'sky-tree-view-toolbar',
+  templateUrl: './sky-tree-view-toolbar.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyTreeViewToolbarComponent {
+
+  @Input()
+  public treeModel: TreeModel;
+
+  @Input()
+  public options: SkyTreeViewOptions;
+
+  public selectAll(): void {
+    this.treeModel.doForAll((node: TreeNode) => {
+      const selectable = !node.hasChildren || (node.hasChildren && !this.options.leafNodeSelectionOnly);
+      if (selectable) {
+        node.setIsSelected(true);
+      }
+    });
+  }
+
+  public deselectAll(): void {
+    this.treeModel.doForAll((node: TreeNode) => {
```

Comment:
It feels a little strange that the toolbar would be messing with the view, directly. Could this be a simple event emitter instead, that the parent component listens to, and then makes changes to its own view?

Reply:
Yeah, that's a good thought. Let me see what I can do.

Reply:
Ok, I've removed the need for the treemodel and utilized emitters so the root `sky-tree-view` component can handle the heavy-lifting. This feels a lot better thanks!